### ISSUE TITLE
Error during upload: the offset is lower than the number of bytes written

### ISSUE THREAD

**Issue by edulodgify** [google-github-actions/upload-cloud-storage#362]
### TL;DR

An unexpected Error: the offset is lower than the number of bytes written.  appears when we upload files.

### Expected behavior

Uploading the files always works

### Observed behavior

we have been observing this behavior where pipelines randomly fail since April with the mentioned error, if we re-run those pipelines they work perfectly

### Action YAML

```yaml
- name: 📤 Update commit build on Google Cloud
        uses: google-github-actions/upload-cloud-storage@v2
        with:
          path: ./build
          destination: ${{ env.GCP_BUCKET_NAME }}/${{ env.GCP_BUCKET_PATH }}/${{ github.sha }}
          parent: false
```

### Log output

```text
Error: google-github-actions/upload-cloud-storage failed with: the offset is lower than the number of bytes written. The server has 0 bytes and while 10 bytes has been uploaded - thus 10 bytes are missing. Stopping as this could result in data loss. Initiate a new upload to continue.
```

### Additional information

It seems that this only happens when we are using the managed runners, with self-hosted we have not observed this problem, but as I said it happens randomly so it may or may not happen in the future

**Comment by sethvargo** [comment]
Hi @edulodgify - this is almost always some kind of transient network, firewall, or connectivity issue that we have no control over. I'm not sure if there's anything we can do in the GitHub Action to make it better.

**Comment by edulodgify** [comment]
Ok, thanks for your help @sethvargo
